src/test/java/teammates/test/cases/action/StudentCourseJoinAuthenticatedActionTest.java

```diff
@@ -279,4 +282,34 @@ public void testExecuteAndPostProcess() throws Exception {
     protected StudentCourseJoinAuthenticatedAction getAction(String... params) {
         return (StudentCourseJoinAuthenticatedAction) gaeSimulation.getActionObject(getActionUri(), params);
     }
+
+    @Override
+    @Test
+    protected void testAccessControl() throws Exception {
+
+        StudentAttributes unregStudent1 = dataBundle.students.get("student1InUnregisteredCourse");
```

Comment:
You can use `getTypicalDataBundle()` to get the desired student in that data set instead of adding it to `StudentCourseJoinAuthenticatedTest.json`.